assets/js/interactivity/directives.js

```diff
@@ -28,79 +42,96 @@ export default () => {
 	directive(
 		'context',
 		( {
-			directives: {
-				context: { default: context },
-			},
+			directives: { context },
 			props: { children },
-			context: inherited,
+			context: inheritedContext,
 		} ) => {
-			const { Provider } = inherited;
-			const inheritedValue = useContext( inherited );
-			const value = useMemo( () => {
-				const localValue = deepSignal( context );
-				mergeDeepSignals( localValue, inheritedValue );
-				return localValue;
-			}, [ context, inheritedValue ] );
-
-			return <Provider value={ value }>{ children }</Provider>;
+			return context.reduceRight(
+				( toReturn, { namespace, value: newContext } ) => {
+					const { Provider } = inheritedContext;
+					const inheritedValue = useContext( inheritedContext );
+					const currentValue = useRef( deepSignal( {} ) );
+					currentValue.current = useMemo( () => {
+						const newValue = deepSignal( {
+							[ namespace ]: newContext,
+						} );
+						mergeDeepSignals( newValue, inheritedValue );
+						mergeDeepSignals(
+							currentValue.current,
+							newValue,
+							true
+						);
+						return currentValue.current;
+					}, [ newContext, inheritedValue ] );
+
+					return (
+						<Provider value={ currentValue.current }>
+							{ toReturn }
+						</Provider>
+					);
+				},
+				[ children ]
+			);
 		},
 		{ priority: 5 }
 	);
 
-	// data-wc-effect--[name]
-	directive( 'effect', ( { directives: { effect }, context, evaluate } ) => {
-		const contextValue = useContext( context );
-		Object.values( effect ).forEach( ( path ) => {
-			useSignalEffect( () => {
-				return evaluate( path, { context: contextValue } );
+	// data-wc-body
+	directive( 'body', ( { props: { children } } ) => {
+		return createPortal( children, document.body );
+	} );
+
+	// data-wc-watch--[name]
+	directive( 'watch', ( { directives: { watch }, evaluate } ) => {
+		watch.forEach( ( entry ) => {
+			useSignalEffect( async () => {
+				const result = evaluate( entry );
+				return await result;
```

Comment:
What is the logic to use an async function here?

Reply:
I'm not sure, to be honest. I simply preserved the current logic, and I don't remember who wrote it. 🤔

PS: I think it was you when you replaced `effect` with `watch`. 😄 c467f35fde5abaa5c16b3d5696f10dd12cddc003

Reply:
Oh, haha, sorry. I don't know what I was thinking about (generators?), but I think it's wrong. We should return the result of evalute directly.